src/connect/wrapMapToProps.js

```diff
@@ -58,7 +58,7 @@ export function wrapMapToPropsFunc(mapToProps, methodName) {
         props = proxy(stateOrDispatch, ownProps)
       }
 
-      if (process.env.NODE_ENV !== 'production') 
+      if (process.env.NODE_ENV !== 'production' && !bypassVerifyPlainObject)
```

Comment:
I think that if you are going to "bypass" this verification then you should check before that the `mapStateToProps` object is valid, like [I'm doing in the code of my PR](https://github.com/reactjs/react-redux/pull/724/files#diff-ed7b415fc4798ed5c269487e2477aef2R20). 😄